pkg/apis/v1/nodepool_conversion_test.go

```diff
@@ -0,0 +1,532 @@
+/*
```

Comment:
I'm not planning to look at this file since I expect it to be fairly similar to the NodeClaim conversion test. I'd mainly just apply the same comments that I added over there into this test file